server/src/main/java/org/apache/iotdb/db/query/reader/universal/IterateReader.java

```diff
@@ -42,19 +43,13 @@ public IterateReader(int readerSize) {
 
   @Override
   public boolean hasNext() throws IOException {
-
     if (curReaderInitialized && currentSeriesReader.hasNext()) {
       return true;
-    } else {
-      curReaderInitialized = false;
     }
-
-    while (nextSeriesReaderIndex < readerSize) {
-      boolean isConstructed = constructNextReader(nextSeriesReaderIndex++);
-      if (isConstructed && currentSeriesReader.hasNext()) {
-        curReaderInitialized = true;
-        return true;
-      }
+    curReaderInitialized = false;
+    if (hasNextChunk()) {
+      curReaderInitialized = true;
+      return currentSeriesReader.hasNext();
```

Comment:
Same above. The if statement should be while statement.
And as the former logic, you can set the curReaderInitialized to true only if currentSeriesReader.hasNext() return true.